src/parser/binary-operation-parser.ts

```diff
@@ -133,7 +133,11 @@ export function parseFilterList(
   const combine = combinator === 'and' ? AndNode.create : OrNode.create
 
   if (list.length === 0) {
-    return ValueNode.createImmediate(combinator === 'and')
+    return BinaryOperationNode.create(
+      ValueNode.createImmediate(1),
+      OperatorNode.create('='),
+      ValueNode.createImmediate(combinator === 'and' ? 1 : 0)
+    )
```

Comment:
`true` and `false` literals are not allowed in mssql, `1 = 1` and `1 = 0` instead work in all the engines.